pkg/flow/adapter/dataweavetransformation/adapter.go

```diff
@@ -96,49 +115,97 @@ func (a *dataweaveTransformAdapter) Start(ctx context.Context) error {
 
 func (a *dataweaveTransformAdapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
 	var err error
-	var tmpfile *os.File
-	path := "/tmp/dw"
-	dwFolder := path + "/custom"
+	var tmpFile *os.File
+	var inputData []byte
+	var spell string
+	var inputContentType string
+	var outputContentType string
 
-	switch a.incomingContentType {
-	case "application/json":
-		if !isValidJSON(event.Data()) {
-			return a.replier.Error(&event, targetce.ErrorCodeRequestParsing,
-				errors.New("invalid Json"), nil)
+	err = validateContentType(event.DataContentType(), event.Data())
+	if err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New(err.Error()), nil)
+	}
+
+	req := &DataWeaveTransformationStructuredRequest{}
```

Comment:
Will the `DataWeaveTransformationStructuredRequest` be the required cloudevent payload format going forward, or will it still be possible to pass just the data to be translated?  If the latter, would suggest migrating the `spellOverride` check up.

Reply:
It will allow the both ways, but I've added some checks with the DataWeaveTransformationStructuredRequest that it needs to be before the spellOverride check